lib/Errors.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright 2018 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+class CustomError extends Error {
+  constructor(message) {
+    super(message);
+    this.name = this.constructor.name;
```

Comment:
This will break if the code gets minified in some way. It is better to assign a string constant in each of the subclasses.

Reply:
> This will break if the code gets minified in some way. It is better to assign a string constant in each of the subclasses.

We don't assume code minification/transpilation usecases, and we already fail miserably with some tools like babel. Does it hurt you? If yes, please file an issue so that your concerns are tracked.